go/util/http_helper.go

```diff
@@ -78,6 +78,9 @@ func Get(ctx context.Context, url string) (*http.Response, error) {
 }
 
 func constructURL(base, path, prefix string) (*url.URL, error) {
+	if !strings.HasPrefix(path, prefix) {
```

Comment:
Keep the path-parsing code together by either moving this down next to the other strings.HasPrefix check on line 89, or moving that one, plus the following url.Parse, up here.